### DISCUSSION TITLE
CI pipeline validations and inferences

### DISCUSSION THREAD

**Discussion by justin2004** [CommonCoreOntology/CommonCoreOntologies#517] [Ontology Terms and Class Schema]
I noticed in [here](https://github.com/CommonCoreOntology/CommonCoreOntologies/tree/develop/.github/deployment/sparql) we have SPARQL validations but I don't think the execution of them is automated. Would that be a welcome feature if I automated that?
Also it might be nice to include the execution of a reasoner so the CI pipeline can fail if we have an inconsistent ontology or malformed RDF files.

**Comment by neilotte** [comment]
@johnbeve

**Comment by johnbeve** [comment]
@justin2004 One of the QC queries has a syntax error that's blocking automated execution during builds. Working to fix that here (https://github.com/CommonCoreOntology/CommonCoreOntologies/pull/519). 

You can see here (https://github.com/CommonCoreOntology/CommonCoreOntologies/actions/runs/10863650645/job/30148040122) outputs of a previous build where the QCs are executing as well as where HermiT is being leveraged to check for consistency. 

Which is to say, absolutely good ideas :)

**Reply by justin2004** [reply]
ah, good!
